### DISCUSSION TITLE
Expire policy for rabbitmq queue if there is not message available

### DISCUSSION THREAD

**Discussion by rajburnwal07** [rabbitmq/rabbitmq-server#11575] [Questions] [answered]
Hey Team, I want to create a expiry policy where queue will be removed if there is no message available for expiry specified. Is it possible by any chance? As per documentation expiry policy will work if there is no consumer available for that queue.

Ref: https://www.rabbitmq.com/docs/ttl#queue-ttl 

You can refer below for expiry policy I created.

apiVersion: rabbitmq.com/v1beta1
kind: Policy
metadata:
  name: policy-ttl-queue
  namespace: rabbitmq
spec:
  name: "policy-demo-ttl-queue"
  vhost: "/"  
  pattern: "^[a-zA-Z0-9_]+-[0-9_]+-[0-9_]+$"
  applyTo: all
  priority: 0
  definition:
    expires: 86400000
  rabbitmqClusterReference:
    name: rabbitmq
    
 Please help me out if there is a way to do it where queue TTL can be set on condition for zero messages?
 
 Thanks in advance.

**Comment by lukebakken** [comment, marked as answer]
> if there is a way to do it where queue TTL can be set on condition for zero messages?

No, this is not supported.

**Reply by rajburnwal07** [reply]
Okay thanks @lukebakken can we do it through curl request by any chance?

Like I read it somewhere its doable through below curl request. could you pls confirm?

`curl -XDELETE https://USERNAME:PASSWORD@host/api/queues/VHOST/QUEUE_NAME -G -d 'if-empty=true' -d 'if-unused=true'`

**Reply by lukebakken** [reply]
> Like I read it somewhere its doable through below curl request. could you pls confirm?

This is something you can _EASILY_ confirm yourself.
